# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/ImageIndex.cpp
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/ImageIndex.cpp
--- a/gfx/angle/checkout/src/libANGLE/ImageIndex.cpp	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/ImageIndex.cpp	Sat Sep 07 13:38:36 2019 +0000
@@ -68,6 +68,18 @@
         return NonCubeTextureTypeToTarget(type);
     }
 }
+
+bool IsArrayTarget(TextureTarget target)
+{
+    switch (target)
+    {
+        case TextureTarget::_2DArray:
+        case TextureTarget::_2DMultisampleArray:
+            return true;
+        default:
+            return false;
+    }
+}
 }  // anonymous namespace
 
 ImageIndex::ImageIndex()
@@ -164,9 +176,10 @@
     return ImageIndex(TextureType::_3D, levelIndex, layerIndex, 1);
 }
 
-ImageIndex ImageIndex::MakeFromTarget(TextureTarget target, GLint levelIndex)
+ImageIndex ImageIndex::MakeFromTarget(TextureTarget target, GLint levelIndex, GLint depth)
 {
-    return ImageIndex(TextureTargetToType(target), levelIndex, TextureTargetToLayer(target), 1);
+    return ImageIndex(TextureTargetToType(target), levelIndex, TextureTargetToLayer(target),
+                      IsArrayTarget(target) ? depth : 1);
 }
 
 ImageIndex ImageIndex::MakeFromType(TextureType type,